---
title: "Processing_Data"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

#prepare the dataset to have all gene expression of three time course (3, 5, 24 hr) for all samples from IntNormOf192StrStAnd28UCDSS_MeanPrSetSummarAndLog2Import_SubtractGeoMeanOf38Normalizers_05-02-13_Opened08-13-15 partek spreadsheet

```{r}
#setwd("C:/Outcome/R_Outcome_Stroke/Outcome_Stroke")
MyData <- read.csv(file="../../Input_Output/IntNormOf192StrStAnd28UCDSS_MeanPrSetSummarAndLog2Import_SubtractGeoMeanOf38Normalizers_05-02-13_Opened08-13-15_09-30-19.csv", header=TRUE, sep=",", stringsAsFactors = FALSE) #220 54683

#Parameter <- read.csv(file="../../Input_Output/Param_Outcome_01-14-20.csv", header=TRUE, sep=",", stringsAsFactors = FALSE, row.names = 1) #206 9

BloodDrawTime <-read.csv(file="../../Input_Output/BloodDrawTime_03-02-20.csv", header=TRUE, sep=",", stringsAsFactors = FALSE) #220 2

merge<- cbind(MyData,BloodDrawTime) # 220 54685

##Demographic <- read.csv(file="../../Input_Output/50_Demographic_03-02-20.csv", header=TRUE, sep=",", stringsAsFactors = FALSE) #50 238

##merge_v2<- merge.data.frame(merge,Demographic) #50 240

##merge_v3 <- merge.data.frame(merge,MyData, by= "Filename") #171 54692 having GE for all 50 samples for all BloodDrawTime == "3" , "5" ,"24"

##merge_v3_3hr <- (subset(merge_v3, BloodDrawTime == 3))
##rownames(merge_v3_3hr) <-merge_v3_3hr[,1] #57 54692
##merge_v3_3hr_v2 <- merge_v3_3hr[merge_v2$Filename,] #50 54931, including demographics and GE for BloodDrawTime == "3" for the 50 samples from Boryana Trial

##Filename <- read.csv("C:/Outcome/R_Outcome_Stroke/Input_Output/57_Outcome_Filename_03-04-20.csv", header=T) #all 57 samples for outcome project

Filename_v2 <- read.csv("C:/Outcome/R_Outcome_Stroke/Input_Output/50_Outcome_Filename_03-04-20.csv", header=T, stringsAsFactors = FALSE) #all 50 samples for outcome project

##merge_v3_5hr <- (subset(merge_v3, BloodDrawTime == "5")) #57 54692
##rownames(merge_v3_5hr) <-merge_v3_5hr[,1] #57 54692
#x <- gsub("B","24", merge_v2$Filename)
#y <- gsub("-0-","-24-", x)
#merge_v3_5hr_v2 <- merge_v3_5hr[y,] #50 54692 including demographics and GE for BloodDrawTime == "5" for the 50 samples from Boryana Trial

merge_3hr <- (subset(merge, BloodDrawTime == 3)) #57 54685
rownames(merge_3hr) <- merge_3hr[,4]
merge_3hr_v2 <- merge_3hr[Filename_v2$X3.hr,] #50 54685


merge_5hr <- (subset(merge, BloodDrawTime == 5)) #57 54685
rownames(merge_5hr) <- merge_5hr[,4]
merge_5hr_v2 <- merge_5hr[Filename_v2$X5.hr,] #50 54685

merge_24hr <- (subset(merge, BloodDrawTime == 24)) #57 54685
rownames(merge_24hr) <- merge_24hr[,4]
merge_24hr_v2 <- merge_24hr[Filename_v2$X24.hr,] #50 54685

#save
write.table(merge,file = "../../Input_Output/New/220_GE_03-22-20.txt", col.names = T, sep = "\t") #can use this for downstream analysis. ALso use this for adding more parameters
save(merge_3hr_v2, file = "../../Input_Output/RData/New/50_GE_3hr_03-22-20.RData")
save(merge_5hr_v2, file = "../../Input_Output/RData/New/50_GE_5hr_03-22-20.RData")
save(merge_24hr_v2, file = "../../Input_Output/RData/New/50_GE_24hr_03-22-20.RData")

##merge_v3_5hr_v2 <- merge_v3_5hr[Filename_v2$X5.hr,]


##merge_v3_24hr <- (subset(merge_v3, BloodDrawTime == 24)) #57 54692
##rownames(merge_v3_24hr) <-merge_v3_24hr[,1] #57 54692
##merge_v3_24hr_v2 <- merge_v3_24hr[Filename_v2$X24.hr,] #50 54931, including demographics and GE for BloodDrawTime == "24" for the 50 samples from Boryana Trial 

##merge_v4 <- merge.data.frame(merge_v2, MyData, by= "Filename") # 50 54922 having GE for all 50 samples for BloodDrawTime == "3"

```

```{r}
#save
##save(merge_v2, file = "../../Input_Output/RData/50_Demographic_03-02-20.RData")
##save(merge_v3_3hr, file = "../../Input_Output/RData/57_Demographic_GE_3hr_03-02-20.RData")
##save(merge_v3_3hr_v2, file = "../../Input_Output/RData/50_Demographic_GE_3hr_03-02-20.RData")
##save(merge_v3_5hr, file = "../../Input_Output/RData/57_Demographic_GE_5hr_03-02-20.RData")
##save(merge_v3_5hr_v2, file = "../../Input_Output/RData/50_Demographic_GE_5hr_03-02-20.RData")
##save(merge_v3_24hr, file = "../../Input_Output/RData/57_Demographic_GE_24hr_03-02-20.RData")
##save(merge_v3_24hr_v2, file = "../../Input_Output/RData/50_Demographic_GE_24hr_03-02-20.RData")


##write.table(merge_v2,file = "../../Input_Output/50_Demographic_Parameter_03-02-20.txt", col.names = T, sep = "\t") #can use this for downstream analysis. ALso use this for adding more parameters
```

```{r}
#calculate delta for GE coming from different time point
##x<- colnames(merge_v3_3hr_v2) #find number of the transcripts (18:54692)
##y <- colnames(merge_v3_5hr_v2) #find number of the transcripts (18:54692)
##z <-colnames(merge_v3_24hr_v2)  #find number of the transcripts (18:54692)
#make sure these three chr are same
x <- colnames(merge_3hr_v2)
y <- colnames(merge_5hr_v2)
z <- colnames(merge_24hr_v2)

setdiff(x,y)
#character(0)
setdiff(x,z)
#character(0)
setdiff(y,z)
#character(0)

threehrsubtractfivehr <- merge_3hr_v2[,c(9:54683)] - merge_5hr_v2[,c(9:54683)] #50 54675 keep 3hr rowname for subtract
##threehrsubtracttwentyfourhr <- merge_v3_3hr_v2[,c(18:54692)] - merge_v3_24hr_v2[,c(18:54692)] #50 54675 keep 3hr rowname for subtract

fivehrsubtracttwentyfourhr <- merge_5hr_v2[,c(9:54683)] - merge_24hr_v2[,c(9:54683)] #50 54675 Here 5hr rowname for subtract output
#we want to keep the Filename consistent so all the files have the 3hr filename
rownames(fivehrsubtracttwentyfourhr) <- rownames(threehrsubtractfivehr)

threehrsubtracttwentyfourhr <- merge_3hr_v2[,c(9:54683)] - merge_24hr_v2[,c(9:54683)] #50 54675



##threehrsubtractfivehr <- merge_v3_3hr_v2[,c(18:54692)] - merge_v3_5hr_v2[,c(18:54692)] #50 54675 keep 3hr rowname for subtract
##threehrsubtracttwentyfourhr <- merge_v3_3hr_v2[,c(18:54692)] - merge_v3_24hr_v2[,c(18:54692)] #50 54675 keep 3hr rowname for subtract
##fivehrsubtracttwentyfourhr <- merge_v3_5hr_v2[,c(18:54692)] - merge_v3_24hr_v2[,c(18:54692)] #50 54675 keep 5hr rowname for subtract
#since I want to keep all the rownames same, used this code to change the rownames
##rownames(fivehrsubtracttwentyfourhr) <- rownames(threehrsubtracttwentyfourhr) #keep 3hr rowname for subtract

```

```{r}
#save
save(threehrsubtractfivehr, file = "../../Input_Output/RData/New/3hrSubtract5hr_GE_03-23-20.RData")
write.table(threehrsubtractfivehr,file = "../../Input_Output/New/3hrSubtract5hr_GE_03-23-20.csv", col.names = T, sep = "\t")

save(threehrsubtracttwentyfourhr, file = "../../Input_Output/RData/New/3hrSubtract24hr_GE_03-23-20.RData")
write.table(threehrsubtracttwentyfourhr,file = "../../Input_Output/New/3hrSubtract24hr_GE_03-23-20.csv", col.names = T, sep = "\t")

save(fivehrsubtracttwentyfourhr, file = "../../Input_Output/RData/New/5hrSubtract24hr_GE_03-23-20.RData")
write.table(fivehrsubtracttwentyfourhr,file = "../../Input_Output/New/5hrSubtract24hr_GE_03-23-20.csv", col.names = T, sep = "\t")
```


```{r}
#save
##save(threehrsubtractfivehr, file = "../../Input_Output/RData/3hrSubtract5hr_GE_03-04-20.RData")
##write.csv(threehrsubtractfivehr,file = "../../Input_Output/3hrSubtract5hr_GE_03-04-20.csv", col.names = T, sep = "\t")
##save(threehrsubtracttwentyfourhr, file = "../../Input_Output/RData/3hrSubtract24hr_GE_03-04-20.RData")
##write.csv(threehrsubtracttwentyfourhr,file = "../../Input_Output/3hrSubtract24hr_GE_03-04-20.csv", col.names = T, sep = "\t")
##save(fivehrsubtracttwentyfourhr, file = "../../Input_Output/RData/5hrSubtract24hr_GE_03-04-20.RData")
##write.csv(fivehrsubtracttwentyfourhr,file = "../../Input_Output/5hrSubtract24hr_GE_03-04-20.csv") # 

```

```{r}
#Merge the demographic with the Xu file which is sent from Jane (most of the columns are same)
##Xu_HumanStrokeSampleMedicalInfo <- read.csv("C:/Outcome/R_Outcome_Stroke/Input_Output/Xu_HumanStrokeSampleMedicalInfo_12-20-08_03-09-20.csv", stringsAsFactors = FALSE) #50 75
##Xu_HumanStrokeSampleMedicalInfo_sort <- Xu_HumanStrokeSampleMedicalInfo[order(Xu_HumanStrokeSampleMedicalInfo$X206SS_Subject.ID),]
#import the demographic file
Demographic_Parameter <- read.delim("C:/Outcome/R_Outcome_Stroke/Input_Output/50_Demographic_Parameter_03-02-20.txt", stringsAsFactors = FALSE)
##x<- cbind.data.frame(Xu_HumanStrokeSampleMedicalInfo_sort, Demographic_Parameter) #50 316
##write.csv(x,file = "../../Input_Output/50_Demographic_Parameter_Xu_HumanStrokeSampleMedicalInfo_03-04-20.csv") 

```

```{r}
#Exclude Filename with ? for NIHSS outcome
exclude <- c("107-0-2_HGU133_Plus_2.CEL","108-B_HGU133_Plus_2.CEL","263-B_HGU133_Plus_2.CEL","CL52-0-4_redo_HGU133_Plus_2.CEL","CL11-0-6_HGU133_Plus_2.CEL","CL118-0-1_HGU133_Plus_2.CEL","CL123-0-4_HGU133_Plus_2.CEL","CL16-0-3_HGU133_Plus_2.CEL","235-B-1_HGU133_Plus_2.CEL","238-B-1_HGU133_Plus_2.CEL","243-B-4_HGU133_Plus_2.CEL","CL32-0-4_HGU133_Plus_2.CEL","CL34-0-1_HGU133_Plus_2.CEL","97-B-2_HGU133_Plus_2.CEL") #14
```

```{r}
#exlude the Filename with ? for outcome
rownames(Demographic_Parameter) <- Demographic_Parameter$Filename
Demographic_Parameter_low = Demographic_Parameter[!row.names(Demographic_Parameter)%in%exclude,] #36 241
write.csv(Demographic_Parameter_low,file = "../../Input_Output/New/36_Demographic_Parameter_03-23-20.csv")
##write.csv(Demographic_Parameter_low,file = "../../Input_Output/36_Demographic_Parameter_Xu_HumanStrokeSampleMedicalInfo_03-04-11.csv")

#keep all the demographic parameters as one file
Demographic_Parameter_low <- read.csv("C:/Outcome/R_Outcome_Stroke/Input_Output/36_Demographic_Parameter_Xu_HumanStrokeSampleMedicalInfo_03-04-11.csv", row.names=1, stringsAsFactors = FALSE)
Demographic_CLEAR <- read.csv("C:/Outcome/R_Outcome_Stroke/Input_Output/Boryana Stamova 03-11-2020 CLEAR.csv", stringsAsFactors = FALSE)
Demographic_CLEAR_sort <- Demographic_CLEAR[order(Demographic_CLEAR$Bar_Code),]

Demographic_CLEAR$Bar_Code
rownames(Demographic_CLEAR_sort) <- Demographic_CLEAR_sort$Bar_Code
x <- rownames(Demographic_CLEAR_sort)
y <- as.integer(x)

Demographic_Parameter_low_sort <- Demographic_Parameter_low[order(Demographic_Parameter_low$X206SS_Bar_Code),]
include <- Demographic_Parameter_low_sort$X206SS_Bar_Code

Demographic_CLEAR_low_sort <- Demographic_CLEAR_sort[x%in%include,] #36 202 some parameters are only available in this file: cbleed, F4Q7a, F4Q7b, F4Q7c, F4Q7d, Day5Paralyzed, BaseLesionVolumecc, Hour24LesionVolumecc, smokingstatus


#merge two demographic files including duplication for most of the parameters
Demographic_all_sort <-cbind.data.frame(Demographic_Parameter_low_sort,Demographic_CLEAR_low_sort) #36 443
rownames(Demographic_all_sort) <- Demographic_all_sort$Filename
```

```{r}
#save
write.csv(Demographic_all_sort,file = "../../Input_Output/36_Demographic_Parameter_All_03-16-20.csv") #all parameters
```

```{r}
#Split 36 samples into 25 DS and 11 VS
DS_Filename <- c("CL102-0-3_HGU133_Plus_2.CEL.pimg","CL104-0-1_HGU133_Plus_2.CEL.pimg","105-B-6_HGU133_Plus_2.CEL.pimg","112-B-2_HGU133_Plus_2.CEL.pimg","CL130-0-6_HGU133_Plus_2.CEL.pimg","CL131-0-4_HGU133_Plus_2.CEL.pimg","CL132-0-3_HGU133_Plus_2.CEL.pimg","CL135-0-2_HGU133_Plus_2.CEL.pimg","CL138-0-6_HGU133_Plus_2.CEL.pimg","201-B-3_HGU133_Plus_2_2.CEL.pimg","210-B-4_HGU133_Plus_2.CEL.pimg","CL22-0-5_HGU133_Plus_2.CEL.pimg","225-B-2_HGU133_Plus_2.CEL.pimg","236-B-3_HGU133_Plus_2.CEL.pimg","255-B-3_HGU133_Plus_2.CEL.pimg","261-B-3_HGU133_Plus_2.CEL.pimg","262-B-5_HGU133_Plus_2.CEL.pimg","278-B-5_HGU133_Plus_2.CEL.pimg","279-B-3_HGU133_Plus_2.CEL.pimg","287-B-4_HGU133_Plus_2.CEL.pimg","289-B-3_HGU133_Plus_2.CEL.pimg","CL38-0-5_HGU133_Plus_2.CEL.pimg","CL55-0-2_HGU133_Plus_2.CEL.pimg","CL56-0-3_HGU133_Plus_2.CEL.pimg","CL150-0-2_HGU133_Plus_2.CEL.pimg") #25

DS_Filename_v2 <- gsub(".pimg", "", DS_Filename) #remove pimg from end of the filename
  
VS_Filename <- c("100-0-3_HGU133_Plus_2.CEL.pimg","CL113-0-3_HGU133_Plus_2.CEL.pimg","CL134-0-5_HGU133_Plus_2.CEL.pimg", "CL136-0-5_HGU133_Plus_2.CEL.pimg","CL18-0-4_HGU133_Plus_2.CEL.pimg","211-B_HGU133_Plus_2.CEL.pimg","230-B-4_HGU133_Plus_2.CEL.pimg","234-B-3_HGU133_Plus_2.CEL.pimg","266-B-2_HGU133_Plus_2.CEL.pimg","271-B-5_HGU133_Plus_2.CEL.pimg","35-B-6_HGU133_Plus_2.CEL.pimg") #11

VS_Filename_v2 <- gsub(".pimg", "", VS_Filename) #remove pimg from end of the filename

DS_Demographic_all <- Demographic_all_sort[rownames(Demographic_all_sort)%in%DS_Filename_v2,] #25 443
VS_Demographic_all <- Demographic_all_sort[rownames(Demographic_all_sort)%in%VS_Filename_v2,] #11 443
```
  
```{r}
#save
write.csv(DS_Demographic_all,file = "../../Input_Output/25_Demographic_Parameter_All_03-16-20.csv")
save(DS_Demographic_all,file = "../../Input_Output/RData/DS_Demographic_all.RData")

write.csv(VS_Demographic_all,file = "../../Input_Output/11_Demographic_Parameter_All_03-16-20.csv")
save(VS_Demographic_all,file = "../../Input_Output/RData/VS_Demographic_all.RData")
```

```{r}
#subtract
#calculate the subtract and %improvement and paste into 25_Demographic_Parameter_All_03-16-20.csv file, convert "0" as a NIHSS lower into "0.5" to calculate the %improvement
#import
DS_Demographic <- read.csv("C:/Outcome/R_Outcome_Stroke/Input_Output/25_Demographic_Parameter_All_03-16-20.csv", row.names=1, stringsAsFactors = FALSE) #25 463 
```

```{r}
#load files including subtract for the GE threehrsubtractfivehr
load( "../../Input_Output/RData/New/3hrSubtract5hr_GE_03-23-20.RData") 
threehrsubtractfivehr %>% dim() #50 54675

#merge two dataframe of NIHSS and GE

x <- rownames(DS_Demographic)
DS_GE_threehrsubtractfivehr <- threehrsubtractfivehr[rownames(threehrsubtractfivehr)%in%x,] #25 54675
d <- DS_Demographic
names <- rownames(d)
rownames(d) <- NULL
data <- cbind(names,d) #25 464

c <- DS_GE_threehrsubtractfivehr
names <- rownames(c)
rownames(c) <- NULL
data2 <- cbind(names,c) #25 54676

DS_GE_threehrsubtractfivehr_v2 <- merge(data, data2, by="names") # 25 55139

#save
write.csv(DS_GE_threehrsubtractfivehr_v2,file = "../../Input_Output/New/25_Demographic_Parameter_GE_threehrsubtractfivehr_03-23-20.csv")
save(DS_GE_threehrsubtractfivehr_v2,file = "../../Input_Output/RData/New/25_Demographic_Parameter_GE_threehrsubtractfivehr.RData")
```


```{r}
#load files including subtract for the GE threehrsubtracttwentyfourhr
load(file = "../../Input_Output/RData/New/3hrSubtract24hr_GE_03-23-20.RData")
threehrsubtracttwentyfourhr %>% dim() #50 54675

#merge two dataframe of NIHSS and GE

x <- rownames(DS_Demographic)
DS_GE_threehrsubtracttwentyfourhr <- threehrsubtracttwentyfourhr[rownames(threehrsubtracttwentyfourhr)%in%x,] #25 54675
#d <- DS_Demographic
#names <- rownames(d)
#rownames(d) <- NULL
#data <- cbind(names,d) #25 464

c <- DS_GE_threehrsubtracttwentyfourhr
names <- rownames(c)
rownames(c) <- NULL
data2 <- cbind(names,c) #25 54676

DS_GE_threehrsubtracttwentyfourhr_v2 <- merge(data, data2, by="names") # 25 55139

#save
write.csv(DS_GE_threehrsubtracttwentyfourhr_v2,file = "../../Input_Output/New/25_Demographic_Parameter_GE_threehrsubtracttwentyfourhr_03-23-20.csv")
save(DS_GE_threehrsubtracttwentyfourhr_v2,file = "../../Input_Output/RData/New/25_Demographic_Parameter_GE_threehrsubtracttwentyfourhr.RData")

```


```{r}
#load files including subtract for the GE fivehrsubtracttwentyfourhr
load(file = "../../Input_Output/RData/New/5hrSubtract24hr_GE_03-23-20.RData")
fivehrsubtracttwentyfourhr %>% dim() #50 54675

x <- rownames(DS_Demographic)
DS_GE_fivehrsubtracttwentyfourhr <- fivehrsubtracttwentyfourhr[rownames(fivehrsubtracttwentyfourhr)%in%x,] #25 54675
#d <- DS_Demographic
#names <- rownames(d)
#rownames(d) <- NULL
#data <- cbind(names,d) #25 464

d <- DS_GE_fivehrsubtracttwentyfourhr
names <- rownames(d)
rownames(d) <- NULL
data2 <- cbind(names,d) #25 54676

DS_GE_fivehrsubtracttwentyfourhr_v2 <- merge(data, data2, by="names") # 25 55139

#save
write.csv(DS_GE_fivehrsubtracttwentyfourhr_v2,file = "../../Input_Output/New/25_Demographic_Parameter_DS_GE_fivehrsubtracttwentyfourhr_03-23-20.csv")
save(DS_GE_fivehrsubtracttwentyfourhr_v2,file = "../../Input_Output/RData/New/25_Demographic_Parameter_GE_fivehrsubtracttwentyfourhr.RData")
```


```{r}
#load files including GE for 3hr, 5hr and 24hr with Demographic parameters
#3hr
load(file = "../../Input_Output/RData/New/50_GE_3hr_03-22-20.RData")
merge_3hr_v2 %>% dim() #50 54685

x <- rownames(DS_Demographic)
DS_GE_threehr <- merge_3hr_v2[rownames(merge_3hr_v2)%in%x,] #25 54685 
#d <- DS_Demographic
#names <- rownames(d)
#rownames(d) <- NULL
#data <- cbind(names,d) #25 464

d <- DS_GE_threehr
names <- rownames(d)
rownames(d) <- NULL
data2 <- cbind(names,d) #25 54686


DS_GE_threehr_v2 <- merge(data, data2, by="names") # 25 55149 (x, y in suffix for two files) 

#save
write.csv(DS_GE_threehr_v2,file = "../../Input_Output/New/25_Demographic_Parameter_GE_threehr_03-23-20.csv")
save(DS_GE_threehr_v2,file = "../../Input_Output/RData/New/25_Demographic_Parameter_GE_threehr.RData")
```

```{r}
#5hr
load(file = "../../Input_Output/RData/New/50_GE_5hr_03-22-20.RData")
merge_5hr_v2 %>% dim() #50 54685

#keep the filename of all files same as three hour with "-B-" and "-0-"
merge_5hr_v3 <- merge_5hr_v2
rownames(merge_5hr_v3) <- rownames(merge_3hr_v2)
DS_GE_fivehr <- merge_5hr_v3[rownames(merge_5hr_v3)%in%x,] #25 54685

#x <- rownames(DS_Demographic)
#d <- DS_Demographic
#names <- rownames(d)
#rownames(d) <- NULL
#data <- cbind(names,d) #25 4

d <- DS_GE_fivehr
names <- rownames(d)
rownames(d) <- NULL
data2 <- cbind(names,d) #25 54686


DS_GE_fivehr_v2 <- merge(data, data2, by="names") # 25 55149 (x, y in suffix for two files) #55156

#save
write.csv(DS_GE_fivehr_v2,file = "../../Input_Output/New/25_Demographic_Parameter_GE_fivehr_03-23-20.csv")
save(DS_GE_fivehr_v2,file = "../../Input_Output/RData/New/25_Demographic_Parameter_GE_fivehr.RData")
```

```{r}
#24hr
load(file = "../../Input_Output/RData/New/50_GE_24hr_03-22-20.RData")
merge_24hr_v2 %>% dim() #50 54685

#keep the filename of all files same as three hour with "-B-" and "-0-"
merge_24hr_v3 <- merge_24hr_v2
rownames(merge_24hr_v3) <- rownames(merge_3hr_v2)
DS_GE_twentyfourhr <- merge_24hr_v3[rownames(merge_24hr_v3)%in%x,] #25 54685

#x <- rownames(DS_Demographic)
#d <- DS_Demographic
#names <- rownames(d)
#rownames(d) <- NULL
#data <- cbind(names,d) #25 4

d <- DS_GE_twentyfourhr
names <- rownames(d)
rownames(d) <- NULL
data2 <- cbind(names,d) #25 54686


DS_GE_fivehr_v2 <- merge(data, data2, by="names") # 25 55149 (x, y in suffix for two files) #55156

#save
write.csv(DS_GE_fivehr_v2,file = "../../Input_Output/New/25_Demographic_Parameter_GE_fivehr_03-23-20.csv")
save(DS_GE_fivehr_v2,file = "../../Input_Output/RData/New/25_Demographic_Parameter_GE_fivehr.RData")
```

save(merge_v3_3hr, file = "../../Input_Output/RData/57_Demographic_GE_3hr_03-02-20.RData")
save(merge_v3_3hr_v2, file = "../../Input_Output/RData/50_Demographic_GE_3hr_03-02-20.RData")
save(merge_v3_5hr, file = "../../Input_Output/RData/57_Demographic_GE_5hr_03-02-20.RData")
save(merge_v3_5hr_v2, file = "../../Input_Output/RData/50_Demographic_GE_5hr_03-02-20.RData")
save(merge_v3_24hr, file = "../../Input_Output/RData/57_Demographic_GE_24hr_03-02-20.RData")
save(merge_v3_24hr_v2, file = "../../Input_Output/RData/50_Demographic_GE_24hr_03-02-20.RData"
```





```{r}
#save
#save(merge, file = "../Input/IntNormOf192StrStAnd28UCDSS_MeanPrSetSummarAndLog2Import_SubtractGeoMeanOf38Normalizers_05-02-13_Opened08-13-15_WithTime_09-30-19.Rdata")
#write.table(merge, file ="../Input/MyData_Time.txt", col.names= T, sep="\t")
```

```{r}
#To have delta gene expression for differenet BloodDrawTime 
# merge_3hr <- merge[merge$BloodDrawTime == "3",] #57 54684
# rownames(merge_3hr) = merge_3hr$Filename
# merge_5hr <- merge[merge$BloodDrawTime == "5",] #57 54684
# rownames(merge_5hr) = merge_5hr$Filename
# merge_24hr <- merge[merge$BloodDrawTime == "24",] #57 54684
# rownames(merge_24hr) = merge_24hr$Filename
# #delta from 3hr-5hr
# merge_subtract3hrfrom5hr <- merge_3hr[,9:54683]-merge_5hr[,9:54683] #57 54675
# merge_subtract3hrfrom5hr_NBDS_NBVS <- merge_subtract3hrfrom5hr[c(NBDS$Filename, NBVS$Filename),] #50 54675
# merge_subtract3hrfrom24hr <- merge_3hr[,9:54683]-merge_24hr[,9:54683] #57 54675
# merge_subtract3hrfrom24hr_NBDS_NBVS <- merge_subtract3hrfrom24hr[c(NBDS$Filename, NBVS$Filename),] #50 54675
# merge_subtract5hrfrom24hr <- merge_5hr[,9:54683]-merge_24hr[,9:54683] #57 54675
# merge_subtract5hrfrom24hr_NBDS_NBVS <- merge_subtract5hrfrom24hr[c(NBDS$Filename, NBVS$Filename),] #50 54675
# 
# 
# write.table(merge_subtract3hrfrom5hr_NBDS_NBVS, file = "../Input/merge_subtract3hrfrom5hr.txt", col.names = T, sep = "\t")
# write.table(merge_subtract3hrfrom24hr_NBDS_NBVS, file = "../Input/merge_subtract3hrfrom24hr.txt", col.names = T, sep = "\t")
# write.table(merge_subtract5hrfrom24hr_NBDS_NBVS, file = "../Input/merge_subtract5hrfrom24hr.txt", col.names = T, sep = "\t")

```

```{r}
#find another completed files for only 50 samples for TimeSinceEvent_BloodDraw_Hr == 3
NBDS <- read.csv(file="../../Input_Output/NBDS_NewCorrectedDerivationSet_02-21-13-IntNormOf192StrStAnd28UCDSS_MeanPrSetSummarAndLog2Import_SubtractGeoMeanOf38Normalizers.csv", row.names = 1, header=TRUE, sep=",",stringsAsFactors=FALSE ) #35 54918 
NBDS[NBDS == "?"] <- NA #convert ? in to NA
NBVS <- read.csv(file="../../Input_Output/NBVS_NewCorrectedDerivationSet_02-21-13-IntNormOf192StrStAnd28UCDSS_MeanPrSetSummarAndLog2Import_SubtractGeoMeanOf38Normalizers.csv", header=TRUE, row.names =1, sep=",", stringsAsFactors = FALSE) # 15 54915
# NBVS[NBVS == "?"] <- NA #convert ? in to NA
```

```{r}
#To see the number of the samples for each NIHSS scales
NBDS_Base_6 <- subset(NBDS, NIHSS_Base == 6) #8 samples
# [1] "CL130-0-6_HGU133_Plus_2.CEL.pimg" "CL131-0-4_HGU133_Plus_2.CEL.pimg" "CL132-0-3_HGU133_Plus_2.CEL.pimg"
# [4] "201-B-3_HGU133_Plus_2_2.CEL.pimg" "225-B-2_HGU133_Plus_2.CEL.pimg"   "236-B-3_HGU133_Plus_2.CEL.pimg"  
# [7] "261-B-3_HGU133_Plus_2.CEL.pimg"   "CL56-0-3_HGU133_Plus_2.CEL.pimg"

table(NBDS$NIHSS_Base)
#NIHSS  6  7  9 10 11 12 13 14 15 16 17 19 20 21 23 26 27 31 
#samples  8  2  1  2  1  3  1  1  2  2  3  2  1  1  1  1  1  2

table(NBDS$X206SS_NIHSS_2hr)
 # 1 10 11 12 13 17 19  2 20 21 22 24 29  3 31  4  5  6  7  8  9 
 # 1  1  2  2  1  2  1  3  1  1  1  1  1  1  1  2  3  1  3  2  2

table(NBDS$X206SS_NIHSS_24hr)
# 0  1 10 11 13 15 17 19  2 21 23 24  3 33  5  6  7  8 
#  1  3  3  2  2  1  3  1  1  1  1  1  5  1  4  1  2  1 

table(NBDS$X206SS_NIHSS_5day)
 # 0  1  3  4  6  7  8  9 11 15 16 19 20 26 28 31 33 34 40 
 # 4  6  2  4  3  1  3  1  1  1  1  1  1  1  1  1  1  1  1 

table(NBDS$Day90NIH)
 # 0  1 10  2  3  4  6  7  8  9 
 # 6  2  1  7  4  1  1  1  1  1

#NBVS
table(NBVS$NIHSS_Base)
 # 6  7  8 10 12 13 14 15 17 20 25 
 # 4  1  1  1  1  2  1  1  1  1  1 

table(NBVS$X206SS_NIHSS_2hr)
 # 0  1  4  5  6  8  9 13 15 23 
 # 1  1  1  2  2  2  1  2  2  1 

table(NBVS$X206SS_NIHSS_24hr)
 # 1  2  3  4  6  7  8 10 14 21 
 # 2  1  2  1  2  1  1  2  2  1 

table(NBVS$X206SS_NIHSS_5day)
 # 0  1  2  3  4  6  7  8 18 19 
 # 2  1  3  1  2  1  1  2  1  1 

table(NBVS$Day90NIH)
 # 0  1 19  4  8 
 # 6  2  1  1  2 


```

```{r}
#calculate delta for different NIHSS time-point for NBDS 
#2 samples (14, 31) having NA for NBDS$X206SS_NIHSS_2hr, two samples were dead
#1 sample (31) having NA for NBDS$X206SS_NIHSS_24hr
#10 samples(4, 5, 7, 8, 14, 19, 21, 22, 30, 35) having NA for NBDS$Day90NIH, 4, 5, 7, 8, 14, 21, 30, 35 were dead
NBDS$expdays
#  [1] NA   NA   NA   "5"  "6"  NA   "7"  "30" NA   NA   NA   NA   NA   "6"  NA   NA   NA   NA   NA   NA   "18"
# [22] NA   NA   NA   NA   NA   NA   NA   NA   "54" NA   NA   NA   NA   "14"


NBDS$Day90NIH <- as.integer(NBDS$Day90NIH)
NBDS$SubtractNIHSSbasefromNIHSS2hr <- NBDS$X206SS_NIHSS_Base - NBDS$X206SS_NIHSS_2hr
NBDS$SubtractNIHSSbasefromNIHSS24hr <- NBDS$X206SS_NIHSS_Base - NBDS$X206SS_NIHSS_24hr
NBDS$SubtractNIHSSbasefromNIHSS5day <- NBDS$X206SS_NIHSS_Base - NBDS$X206SS_NIHSS_5day
NBDS$SubtractNIHSSbasefromDay90NIH <- NBDS$X206SS_NIHSS_Base - NBDS$Day90NIH


NBDS$SubtractNIHSS2hrfromNIHSS24hr <- NBDS$X206SS_NIHSS_2hr - NBDS$X206SS_NIHSS_24hr
NBDS$SubtractNIHSS2hrfromNIHSS5day <- NBDS$X206SS_NIHSS_2hr - NBDS$X206SS_NIHSS_5day
NBDS$SubtractNIHSS2hrfromDay90NIH <- NBDS$X206SS_NIHSS_2hr - NBDS$Day90NIH

NBDS$SubtractNIHSS24hrfromNIHSS5day <- NBDS$X206SS_NIHSS_24hr - NBDS$X206SS_NIHSS_5day
NBDS$SubtractNIHSS24hrfromDay90NIH <- NBDS$X206SS_NIHSS_24hr - NBDS$Day90NIH

NBDS$SubtractNIHSS5dayfromDay90NIH <- NBDS$X206SS_NIHSS_5day - NBDS$Day90NIH
```

```{r}
#calculate delta for different NIHSS time-point for NBVS
#3 samples (2, 11, 15 )having NA for NBVS$Day90NIH, sample 15 was dead
NBVS$expdays
# [1] NA   NA   NA   NA   NA   NA   NA   NA   NA   NA   NA   NA   NA   NA   "14"

NBVS$Day90NIH <- as.integer(NBVS$Day90NIH)
NBVS$SubtractNIHSSbasefromNIHSS2hr <- NBVS$X206SS_NIHSS_Base - NBVS$X206SS_NIHSS_2hr
NBVS$SubtractNIHSSbasefromNIHSS24hr <- NBVS$X206SS_NIHSS_Base - NBVS$X206SS_NIHSS_24hr
NBVS$SubtractNIHSSbasefromNIHSS5day <- NBVS$X206SS_NIHSS_Base - NBVS$X206SS_NIHSS_5day
NBVS$SubtractNIHSSbasefromDay90NIH <- NBVS$X206SS_NIHSS_Base - NBVS$Day90NIH


NBVS$SubtractNIHSS2hrfromNIHSS24hr <- NBVS$X206SS_NIHSS_2hr - NBVS$X206SS_NIHSS_24
NBVS$SubtractNIHSS2hrfromDay90NIH <- NBVS$X206SS_NIHSS_2hr - NBVS$Day90NIH

NBVS$SubtractNIHSS24hrfromNIHSS5day <- NBVS$X206SS_NIHSS_24hr - NBVS$X206SS_NIHSS_5day
NBVS$SubtractNIHSS24hrfromDay90NIH <- NBVS$X206SS_NIHSS_24hr - NBVS$Day90NIH

NBVS$SubtractNIHSS5dayfromDay90NIH <- NBVS$X206SS_NIHSS_5day - NBVS$Day90NIH
```

```{r}
#save the modified NBVS and NBDS
save(NBDS, NBVS, file = "../Input/RData/NBDS_NBVS.RData")

```

